packages/fx-core/src/plugins/resource/bot/v2/plugin.ts

```diff
@@ -21,50 +11,37 @@ import {
 } from "@microsoft/teamsfx-api/build/v2";
 import { scaffold } from "./scaffold";
 import * as utils from "../utils/common";
-import { HostTypeTriggerOptions } from "../question";
 import path from "path";
 import { AzureHostingFactory } from "../../../../common/azure-hosting/hostingFactory";
-import { isBotNotificationEnabled } from "../../../../common";
-import { AzureSolutionQuestionNames } from "../../../solution/fx-solution/question";
-import {
-  QuestionNames,
-  TemplateProjectsConstants,
-  TemplateProjectsScenarios,
-  TriggerTemplateScenarioMappings,
-} from "../constants";
-import { PluginActRoles } from "../enums/pluginActRoles";
-import {
-  BotTrigger,
-  PluginBot,
-  QuestionBotScenarioToPluginActRoles,
-  CommonStrings,
-  Commands,
-} from "../resources/strings";
-import { CodeTemplateInfo } from "./interface/codeTemplateInfo";
+import { PluginBot, CommonStrings, Commands } from "../resources/strings";
 import { CommandExecutionError } from "../errors";
-import { BicepConfigs, HostType } from "../../../../common/azure-hosting/interfaces";
+import { BicepConfigs, ServiceType } from "../../../../common/azure-hosting/interfaces";
 import { mergeTemplates } from "../../../../common/azure-hosting/utils";
 import { getActivatedV2ResourcePlugins } from "../../../solution/fx-solution/ResourcePluginContainer";
 import { NamedArmResourcePluginAdaptor } from "../../../solution/fx-solution/v2/adaptor";
 import { ResourcePlugins } from "../../../../common/constants";
+import { runtimeMap } from "./mapping";
+import {
+  getTemplateInfos,
+  resolveHostType,
+  resolveProgrammingLanguage,
+  resolveServiceType,
+} from "./common";
 
 export class TeamsBotV2Impl {
   async scaffoldSourceCode(ctx: Context, inputs: Inputs): Promise<Result<Void, FxError>> {
-    const workingPath = path.join(inputs.projectPath ?? "", "bot");
-    const hostTypeTriggers = inputs[QuestionNames.BOT_HOST_TYPE_TRIGGER];
-    let hostType;
-    if (Array.isArray(hostTypeTriggers)) {
-      const hostTypes = hostTypeTriggers.map(
-        (item) => HostTypeTriggerOptions.find((option) => option.id === item)?.hostType
-      );
-      hostType = hostTypes ? hostTypes[0] : undefined;
+    let workingPath = inputs.projectPath ?? "";
+    const lang = resolveProgrammingLanguage(ctx);
+    if (lang === "csharp") {
```

Comment:
1. Better to ensure if there's some enum can be used to replace the plain string 'csharp'.
2. I guess we should use `!==` here?